core/renderers/thrasos/info.js

```diff
@@ -134,43 +135,43 @@ Blockly.thrasos.RenderInfo.prototype.getInRowSpacing_ = function(prev, next) {
 
   // Between inputs and the end of the row.
   if (prev.isInput && !next) {
-    if (prev.isExternalInput()) {
+    if (Blockly.blockRendering.Types.isExternalInput(prev)) {
       return this.constants_.NO_PADDING;
-    } else if (prev.isInlineInput()) {
+    } else if (Blockly.blockRendering.Types.isInlineInput(prev)) {
       return this.constants_.LARGE_PADDING;
-    } else if (prev.isStatementInput()) {
+    } else if (Blockly.blockRendering.Types.isStatementInput(prev)) {
       return this.constants_.NO_PADDING;
     }
   }
 
   // Spacing between a non-input and an input.
-  if (!prev.isInput && next.isInput) {
+  if (!prev.isInput && next && next.isInput) {
```

Comment:
Should isInput be part of the type checker?  If not, why not?

Reply:
Hmm, looking closer at this. it looks like Blockly.blockRendering.InputConnection extends measurable but doesn't update its type. Instead it relies on isInput to determine if a measurable is of an input type. 
With this, we can have it set it's type to ``INPUT`` and then each subclass can extend the type to also include the specific type enum flag.
I like that solution better, can fix if that makes sense.

Reply:
Yup, that's what I expected based on the pattern elsewhere.

Though InputConnection extends Connection, which extends Measurable, rather than extending Measurable directly.

Reply:
Yeah so in this design an InputConnection is an INPUT and a CONNECTION. do you see any problem with that?